File: customer-app/lib/features/marketplace/models/fulfillment_journey.dart
class FulfillmentJourney {
  final String id;
  final String type;
  final Map<String, dynamic>? config;
  final String sellerId;

  FulfillmentJourney({
    required this.id,
    required this.type,
    this.config,
    required this.sellerId,
  });

  factory FulfillmentJourney.fromJson(Map<String, dynamic> json) {
    return FulfillmentJourney(
      id: json['id'] as String,
      type: json['type'] as String,
      config: json['config'] as Map<String, dynamic>?,
      sellerId: json['seller_id'] as String,
    );
  }
}

File: mercur-store/backend/src/modules/fulfillment/models/fulfillment-journey.ts
import { model } from "@medusajs/framework/utils"

export const FulfillmentJourney = model.define("fulfillment_journey", {
    id: model.id().primaryKey(),
    type: model.enum(["shopify", "form", "voucher", "whatsapp", "manual"]).default("manual"),
    config: model.json().nullable(),
    seller_id: model.text().searchable(),
})
